### PR TITLE
Limit sql_mode changes to removal of "ONLY_FULL_GROUP_BY"

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

Currently, `sql_mode` is set to empty by default to ensure compatibility with MySQL 5.7+. I guess the main problem was the handling of the `ONLY_FULL_GROUP_BY` mode (I tried to activate it and saw that handling it would be a real pain). So I propose to remove only this problematic mode, in order to be able to detect errors/warnings triggered by other modes.

The only problem I found was that some comment fields were not nullable and had no default value:
```
In toolbox.class.php line 2299:
                                                               
  Field 'comment' doesn't have a default value                 
  Error execution statement in table glpi_rulerightparameters  
  Array                                                        
  (                                                            
      [id] => 1                                                
      [name] => (LDAP)Organization                             
      [value] => o                                             
  )                                                            
                                                               

glpi:database:install [-H|--db-host [DB-HOST]] [-d|--db-name DB-NAME] [-p|--db*** [-P|--db-port [DB-PORT]] [-u|--db-user DB-USER] [-r|--reconfigure] [-L|--default-language [DEFAULT-LANGUAGE]] [-f|--force]
```

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -127,8 +127,8 @@ jobs:
           - {php-version: "7.3", db-image: "mariadb:10.5", always: false}
           - {php-version: "7.4", db-image: "mariadb:10.5", always: false}
           - {php-version: "7.4", db-image: "mariadb:10.1", always: false}
-          - {php-version: "7.4", db-image: "mysql:5.6", always: false}
-          - {php-version: "7.4", db-image: "mysql:8.0", always: false}
+          - {php-version: "7.4", db-image: "mysql:5.6", always: true}
+          - {php-version: "7.4", db-image: "mysql:8.0", always: true}
```
Comment (cedric-anne) [inline .github/workflows/ci.yml:131]: This should not be included in PR, will be removed once validated.

**Review by orthagh — APPROVED** [review]

File: inc/dbmysql.class.php [hunk 2]
```diff
@@ -214,8 +214,13 @@ function connect($choice = null) {
...
          $this->dbh->options(MYSQLI_OPT_INT_AND_FLOAT_NATIVE, true);
 
-         if (GLPI_FORCE_EMPTY_SQL_MODE) {
+         if (defined('GLPI_FORCE_EMPTY_SQL_MODE')) {
+            Toolbox::deprecated('Usage of GLPI_FORCE_EMPTY_SQL_MODE has been deprecated');
+         }
+         if (defined('GLPI_FORCE_EMPTY_SQL_MODE') && GLPI_FORCE_EMPTY_SQL_MODE) {
             $this->dbh->query("SET SESSION sql_mode = ''");
+         } else if (GLPI_DISABLE_ONLY_FULL_GROUP_BY_SQL_MODE) {
+            $this->dbh->query("SET SESSION sql_mode = (SELECT REPLACE(@@sql_mode, 'ONLY_FULL_GROUP_BY', ''))");
```
Comment (trasher) [inline inc/dbmysql.class.php:223]:
```suggestion
         if (defined('GLPI_FORCE_EMPTY_SQL_MODE') && GLPI_FORCE_EMPTY_SQL_MODE) {
            Toolbox::deprecated('Usage of GLPI_FORCE_EMPTY_SQL_MODE has been deprecated');
            $this->dbh->query("SET SESSION sql_mode = ''");            
         } else if (GLPI_DISABLE_ONLY_FULL_GROUP_BY_SQL_MODE) {
            $this->dbh->query("SET SESSION sql_mode = (SELECT REPLACE(@@sql_mode, 'ONLY_FULL_GROUP_BY', ''))");
```

File: install/mysql/glpi-empty.sql [hunk 3]
```diff
@@ -151,7 +151,7 @@ CREATE TABLE `glpi_apiclients` (
   `app_token` VARCHAR(255) DEFAULT NULL,
   `app_token_date` timestamp NULL DEFAULT NULL,
   `dolog_method` TINYINT NOT NULL DEFAULT '0',
-  `comment` TEXT,
+  `comment` text COLLATE utf8_unicode_ci,
```
Comment (trasher) [inline install/mysql/glpi-empty.sql:154]: This one should be added in migration aswell, no? I do not knwon what was used when nothing was define; (but I won't bet on the one we'd like :D

Comment (trasher) [reply install/mysql/glpi-empty.sql:154]: (same for `glpi_appliance*`)

File: inc/dbmysql.class.php (hunk 2 above)
Comment (cedric-anne) [reply inc/dbmysql.class.php:223]: I wanted to have a deprectation even when `GLPI_FORCE_EMPTY_SQL_MODE` is set to `0`.

Comment (trasher) [reply inc/dbmysql.class.php:223]: Why would we care about that?

**Review by trasher — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (trasher) [reply .github/workflows/ci.yml:131]: You can proceed ;)

Comment (cedric-anne) [reply .github/workflows/ci.yml]: Done.

File: inc/dbmysql.class.php [hunk 4]
```diff
@@ -214,8 +214,11 @@ function connect($choice = null) {
          // force mysqlnd to return int and float types correctly (not as strings)
          $this->dbh->options(MYSQLI_OPT_INT_AND_FLOAT_NATIVE, true);
 
-         if (GLPI_FORCE_EMPTY_SQL_MODE) {
-            $this->dbh->query("SET SESSION sql_mode = ''");
+         if (defined('GLPI_FORCE_EMPTY_SQL_MODE') && GLPI_FORCE_EMPTY_SQL_MODE) {
+            Toolbox::deprecated('Usage of GLPI_FORCE_EMPTY_SQL_MODE has been deprecated');
+            $this->dbh->query("SET SESSION sql_mode = ''");
```
Comment (trasher) [inline inc/dbmysql.class.php:219]:
```suggestion
            $this->dbh->query("SET SESSION sql_mode = ''");
```

**Pull request merged**